kubeflow/common/iap-ingress/base/config-map.yaml

```diff
@@ -125,65 +125,14 @@ data:
-        BACKEND_NAME=$(echo $BACKENDS | grep -o "k8s-be-${NODE_PORT}--[0-9a-z]\+")
-        echo "backend name is ${BACKEND_NAME}"
-        sleep 2
-      done
-
-      while [[ -z ${BACKEND_SERVICE} ]];
-      do BACKEND_SERVICE=$(gcloud --project=${PROJECT} compute backend-services list --filter=name~${BACKEND_NAME} --uri);
-      echo "Waiting for the backend-services resource PROJECT=${PROJECT} BACKEND_NAME=${BACKEND_NAME} SERVICE=${SERVICE}...";
-      sleep 2;
-      done
-
-      while [[ -z ${HEALTH_CHECK_URI} ]];
-      do HEALTH_CHECK_URI=$(gcloud compute --project=${PROJECT} health-checks list --filter=name~${BACKEND_NAME} --uri);
-      echo "Waiting for the healthcheck resource PROJECT=${PROJECT} NODEPORT=${NODE_PORT} SERVICE=${SERVICE}...";
-      sleep 2;
-      done
+      # See: https://cloud.google.com/service-mesh/docs/iap-integration#deploying_the_load_balancer
+      HC_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="status-port")].nodePort}')
+      HC_INGRESS_PATH=$(kubectl -n istio-system get deployments istio-ingressgateway -o jsonpath='{.spec.template.spec.containers[?(@.name=="istio-proxy")].readinessProbe.httpGet.path}')
 
-      echo health check URI is ${HEALTH_CHECK_URI}
-
-      # Since we create the envoy-ingress ingress object before creating the envoy
-      # deployment object, healthcheck will not be configured correctly in the GCP
-      # load balancer. It will default the healthcheck request path to a value of
-      # / instead of the intended /healthz.
-      # Manually update the healthcheck request path to /healthz
-      if [[ ${HEALTHCHECK_PATH} ]]; then
-        # This is basic auth
-        echo Running health checks update ${HEALTH_CHECK_URI} with ${HEALTHCHECK_PATH}
-        gcloud --project=${PROJECT} compute health-checks update http ${HEALTH_CHECK_URI} --request-path=${HEALTHCHECK_PATH}
-      else
-        # /healthz/ready is the health check path for istio-ingressgateway
-        echo Running health checks update ${HEALTH_CHECK_URI} with /healthz/ready
-        gcloud --project=${PROJECT} compute health-checks update http ${HEALTH_CHECK_URI} --request-path=/healthz/ready
-        # We need the nodeport for istio-ingressgateway status-port
-        STATUS_NODE_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="status-port")].nodePort}')
-        gcloud --project=${PROJECT} compute health-checks update http ${HEALTH_CHECK_URI} --port=${STATUS_NODE_PORT}
-      fi      
+      echo Setting BackendConfig healthCheck.port to: ${HC_INGRESS_PORT} and healthCheck.requestPath to ${HC_INGRESS_PATH}
+      kubectl patch backendconfig iap-backendconfig -n ${NAMESPACE} --type json -p '[{"op": "replace", "path": "/spec/healthCheck/port", "value": '${HC_INGRESS_PORT}'}, {"op": "replace", "path": "/spec/healthCheck/requestPath", "value": "'${HC_INGRESS_PATH}'"}]'
```

Comment:
NIT: Can we break down the patch into two commands?

```
kubectl patch backendconfig iap-backendconfig -n ${NAMESPACE} --type json -p '[{"op": "replace", "path": "/spec/healthCheck/port", "value": '${HC_INGRESS_PORT}'}]'
kubectl patch backendconfig iap-backendconfig -n ${NAMESPACE} --type json -p '[{"op": "replace", "path": "/spec/healthCheck/requestPath", "value": "'${HC_INGRESS_PATH}'"}]'
```

This is for easier debugging by separating individual patches, and for readability on the backend updater script.